### ISSUE TITLE
False Positive: "Variable name has already been declared previously without being used"

### ISSUE THREAD

**Issue by klx01** [kalessil/phpinspectionsea#1613]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Extended) 4.0.5 (and previous versions as well) |
| Language level | PHP 7.4 (and previous versions as well)                        |

Current behaviour
--
```php
function foo(bool $a, bool $b, bool $c)
{
    if ($b && $c) {
        // ...
        if ($a) {  // <- variable is used
            $d = 1;
        }
    }
    $a = foobar(); // False Positive - [EA] This variable name has already been declared previously without being used.
}

// if $c is removed from condition, the error disappears
function bar(bool $a, bool $b, bool $c)
{
    if ($b) {
        // ...
        if ($a) {
            $d = 1;
        }
    }
    $a = foobar(); // no error
}

// or if all variables are used in same condition, the error disappears as well
function baz(bool $a, bool $b, bool $c)
{
    if ($b && $c && $a) {
        $d = 1;
    }
    $a = foobar(); // no error
}

// but if assignment is changed from function call to value, the error returns
function baz2(bool $a, bool $b, bool $c)
{
    if ($b && $c && $a) {
        $d = 1;
    }
    $a = true; // False Positive - [EA] This variable name has already been declared previously without being used. 
}
```

Expected behaviour
--
No error in all cases
